src/ui/public/kuery/lib/functions/not.js

```diff
@@ -0,0 +1,38 @@
+import * as ast from '../ast';
+import { nodeTypes } from '../node_types';
+
+export function buildNodeParams(child, serializeStyle = 'operator') {
+  return {
+    arguments: [child],
+    serializeStyle
+  };
+}
+
+export function toElasticsearchQuery(node, indexPattern) {
+  let [ argument ] = node.arguments;
+  if (argument.type === 'literal') {
+    argument = nodeTypes.function.buildNode('is', '*', argument.value);
+  }
+
+  return {
+    bool: {
+      must_not: ast.toElasticsearchQuery(argument, indexPattern)
+    }
+  };
+}
+
+export function toKueryExpression(node) {
+  if (node.serializeStyle !== 'operator') {
+    throw new Error(`Cannot serialize "not" function as "${node.serializeStyle}"`);
+  }
+
+  const [ argument ] = node.arguments;
+  const queryString = ast.toKueryExpression(argument);
+
+  if (argument.function && (argument.function === 'and' || argument.function === 'or')) {
```

Comment:
Forgot to mention: I updated this condition to check for the `function` serializeStyle, so that in the meantime we won't run into the unnecessary wrapping you described.

03ed9220e80bfab126e6d4be90be809cffc13435